third_party/libusb/naclport/src/chrome_usb/chrome-usb-backend.js

```diff
@@ -228,10 +231,43 @@ ChromeUsbBackend.prototype.reportRequestError_ = function(
  */
 ChromeUsbBackend.prototype.reportRequestSuccess_ = function(
     debugRepresentation, promiseResolver, resultArgs) {
+  if (this.fakeNoDevices &&
+      (debugRepresentation.startsWith('chrome.usb.getDevices') ||
+      debugRepresentation.startsWith('chrome.usb.getConfigurations'))) {
+    resultArgs[0] = [];
+  }
   this.logger.fine(
       'Results returned by the ' + debugRepresentation + ' call: ' +
       goog.iter.join(goog.iter.map(resultArgs, debugDump), ', '));
   promiseResolver.resolve(resultArgs);
 };
 
+/**
+ * @param {!chrome.loginState.ProfileType} profileType
+ * @private
+ */
+ChromeUsbBackend.prototype.receiveProfileType_ = function(profileType) {
+  if (chrome.loginState &&
+      profileType === chrome.loginState.ProfileType.USER_PROFILE) {
+    chrome.loginState.onSessionStateChanged.addListener(
+        this.sessionStateChangedListener_.bind(this));
+  }
+};
+
+/**
+ * @param {!chrome.loginState.SessionState} sessionState
+ * @private
+ */
+ChromeUsbBackend.prototype.sessionStateChangedListener_ = function(
+    sessionState) {
+  if (chrome.loginState &&
+      sessionState === chrome.loginState.SessionState.IN_LOCK_SCREEN) {
+    this.logger.info("No longer showing USB devices.");
+    this.fakeNoDevices = true;
+  } else {
+    this.logger.info("Showing USB devices.");
+    this.fakeNoDevices = false;
+  }
+};
+
```

Comment:
As a general feedback for this CL, I'm thinking whether we can structure this new logic a bit better by extracting into a separate class. What we're adding here is a separate piece of functionality that can be isolated from the existing logic in this class that is, basically, responsible for converting the data between the NaCl library format and the chrome.usb format.

Additionally, extracting new code into a separate class will allow testing it easily. As a bonus, this will help us when doing the migration from the chrome.usb API to the other API (WebUSB?), since the logic around the Lock Screen should stay intact regardless of how exactly the USB APIs are implemented.

There are many ways for implementing this. For example, we can say that the ChromeUsbBackend class provides "addHook"/"removeHook" methods, and the hooks are called on every request and response.

But I haven't looked at your other CL yet - you might want to come up with some approach that can be reused in both CLs.